library/src/main/java/com/mapzen/android/MapzenMap.java

```diff
@@ -676,4 +708,26 @@ public void drawTransitRouteLine(@NotNull List<LngLat> points, @Nullable List<Ln
   public void clearTransitRouteLine() {
     overlayManager.clearTransitRouteLine();
   }
+
+  /**
+   * By default all {@link MapData} objects are removed from the map when it is destroyed. To
+   * persist this data such as in the case of orientation changes, use this method.
+   * @param persistOnRecreation persist {@link MapData} across orientation changes
+   */
+  public void setPersistMapData(boolean persistOnRecreation) {
+    overlayManager.setPersistMapData(persistOnRecreation);
+    mapStateManager.setPersistMapState(persistOnRecreation);
+  }
+
+  private void restoreMapState() {
```

Comment:
Based on initial non-scientific testing (using a stopwatch timer) invoking this method seems to double the time it takes before the map is displayed on the screen from ~2 seconds to ~4 seconds which feels _really_ long.

I wonder if there is anything we can do to speed up the process of recreating the map state?

cc @blair1618 @tallytalwar